xgcm/grid.py

```diff
@@ -178,13 +178,22 @@ def __init__(
             # TODO (Julius in #568) full hierarchy of conventions here
             # but override with any user-given options
 
+            # try sgrid parsing
+            if sgrid.assert_valid_sgrid(ds):
+                sgrid_ax_names = sgrid.get_all_axes(ds)
+                parsed_coords = {}
+                for ax_name in sgrid_ax_names:
+                    parsed_coords[ax_name] = sgrid.get_axis_positions_and_coords(
+                        ds, ax_name
+                    )
```

Comment:
Can we encapsule this into a single method? 

Something like this?
```python
ds, sgrid_kwargs = sgrid.parse_metadata(ds)
```

Reply:
So a user would have to manually parse the dataset for sgrid `kwargs` (or comodo, or cf etc.) before passing them to `grid.Grid`?
Or do you still want this to be called as part of `grid.Grid` to incorporate some autoguessing as discussed in https://github.com/xgcm/xgcm/issues/566#issuecomment-1351705458

Reply:
I lean towards making both possible and defaulting to parsing at grid creation. I think this is a good discussion point for the meeting.